enhancements/update/upgrades-blocking-on-ack.md

```diff
@@ -0,0 +1,138 @@
+
+The new configmap `upgrade-ack` will be installed as part of the release in the namespace `openshift-cluster-version`.
+It will initially have no keys since the administrator is expected to add the relevant key as part of acknowledgement.
+
+### User Stories
+
+Clusters upgraded to 4.9 will no longer have access to these [APIs][k8s-1.22-removed-apis].
+This new upgrade blocking mechanism will be used in an attempt to require administrators to read detailed documentation regarding the removed APIs and the possible impact.
+This OCP documentation will list which APIs have been removed and how an administrator checks their cluster for use of those APIs.
+
+4.8.z versions that are upgradeable to 4.9 will contain this enhancement and have the configmap `openshift-cluster-version/upgrade-ack`.
+CVO will set `Upgradeable=false` which in turn blocks cluster upgrades to 4.9 until the administrator has added the key/value pair `ack-4.9-api-removal-upgrade-approved: "true"` to the configmap.
+Until this is done the `Conditions` console page will display an `Upgradeable` condition message directing the administrator to the relevant OCP documentation.
+
+If the administrator wants to upgrade to 4.9 without performing the acknowledgment `force` can be used.
+
+### Implementation Details/Notes/Constraints
+
+### Risks and Mitigations
+
+* A common risk is human error resulting in an incorrect key/value pair being placed in the configmap.
+This type of error will not change `Upgradeable=false` and therefore the administrator will know a mistake has been made since the `Upgradeable` condition message will continue to be displayed.
+CVO will log an error whenever an unrecognized key is found in the configmap.
+* Another risk is that the configmap gets removed.
+This will result in `Upgradeable=false` and the standard `Upgradeable` condition message will be displayed with the additional message that the configmap cannot be found and should be created. (OPTION: Should CVO create the configmap?)
+CVO will log an error whenever there is an issue accessing the configmap.
+
+## Design Details
+
+### Test Plan
+
+* CVO unit tests will be expanded and/or created to test the new logic.
+* e2e upgrade test will be enhanced.
+
+### Graduation Criteria
+
+GA. When it works, we ship it.
+
+#### Dev Preview -> Tech Preview
+
+N/A. This is not expected to be released as Dev Preview.
+
+#### Tech Preview -> GA
+
+N/A. This is not expected to be released as Tech Preview.
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+These keys must be examined for validity for the next minor release and removed if no longer valid. For example, if the given 4.9 API breakage is no longer considered an issue when 4.10 is available the CVO `ack-4.9-api-removal-upgrade-approved` based logic and related documentation must be removed from 4.9.
```

Comment:
i generally agree that the logic can/should be removed (specifically because we don't want a newly installed 4.9 cluster to be blocked from upgrading to 4.10 because the configmap is missing this key), but if we do remove the logic that expects this key, and the admin does not remove the key from the configmap (no reason for them to do so), then isn't the CVO going to start complaining about unknown keys in the configmap?

Reply:
in general i think we want to encourage the admin to leave the previous acks in place, so they serve as a bit of history/context.  (this is also why i think it's important that the key names include the version they apply to)

Reply:
I agree. I do not think we need to delete the configmap . But if admins want to delete it, they should be fine doing that right?

Reply:
Removal of keys no longer required since their name will version scope them.